sys/include/net/fib.h

```diff
@@ -170,8 +178,217 @@ int fib_get_next_hop(kernel_pid_t *iface_id,
 *                  however the required size may change in between calls.
 */
 int fib_get_destination_set(uint8_t *prefix, size_t prefix_size,
-                            fib_destination_set_entry_t *dst_set, size_t* dst_set_size);
+                            fib_destination_set_entry_t *dst_set, size_t *dst_set_size);
+
+/**
+* @brief creates a new source route
+*
+* @param[in, out] fib_sr pointer to store the new created source route pointer
+* @param[in] sr_iface_id the interface ID used for the created source route
```

Comment:
Usually a source route, more specifically the first `next-hop` on it, is learned on a specific interface so it is bound to that interface. 
When you use a source route you want to use the proper interface since else you will talk to the wrong network.